src/App.tsx

```diff
@@ -1,34 +1,77 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import './App.css'
+import './App.css';
+import Navbar from './components/Navbar';
+import Categories from './components/Categories';
+import ProductPage from './components/ProductPage';
+import Cart from './components/Cart';
+import { useState } from 'react';
+
+interface ProductCartItem {
+  id: string;
+  image: string;
+  title: string;
+  price: number;
+  category: string;
+  description: string;
+  quantity: number;
+}
 
 function App() {
-  const [count, setCount] = useState(0)
+  // gett current route
+  const currentPath = window.location.pathname;
+
+  // state variables
+  const [isCartOpen, setIsCartOpen] = useState(false);
+  const [cartProducts, setCartProducts] = useState<ProductCartItem[]>([]);
+  const [cartCount, setCartCount] = useState(0);
```

Comment:
These useStates contain dependent information. Having multiple useStates with subtle redundancy like this opens the door for bugs where they get out of sync. It also adds complexity to keep them in sync. Instead define `cartCount` as a `useMemo` that listens for changes to `cartProducts` and sums all the quantities. 

The technical consequences of this approach is that we have an extra rerender to update cartCount after cartProducts. In practice though we worry less about optimizing rerenders versus maintainable code.

Caring about these small things is completely unnecessary in a small personal project but I offer these nitpicks to inform how we think about this stuff in ADDI.

```suggestion
  const [cartProducts, setCartProducts] = useState<ProductCartItem[]>([])
  const cartCount = useMemo(() => ....
```